python/jiminy_py/src/jiminy_py/plotter.py

```diff
@@ -0,0 +1,74 @@
+# Plot data from a jiminy log file. See inline help for more info.
+
+import argparse
+import numpy as np
+import matplotlib.pyplot as plt
+from fnmatch import filter
+
+from jiminy_py.core import Engine
+
+def main():
+    description_str = "Plot data from a jiminy log file using matplotlib.\n" + \
+                      "Simply specify a list of fields to plot, separated by a colon for plotting on the same subplot.\n" +\
+                      "Example: h1 h2:h3 generates two subplots, one with h1, one with h2 and h3.\n" + \
+                      "Regular expressions can be used. Enter no plot command (only the file name) to view the list of fields available inside the file."
+
+    parser = argparse.ArgumentParser(description = description_str, formatter_class = argparse.RawTextHelpFormatter)
+    parser.add_argument("input", help = "Input logfile.")
+    main_arguments, plotting_commands = parser.parse_known_args()
+
+    # Load log file.
+    log_data, _ = Engine.read_log(main_arguments.input)
```

Comment:
either we should be able to deal with the csv file, or completely remove the support.

Reply:
This is an issue that was present before this PR: only binary  log can be read.

I think  csv is a familiar format for many people so it's nice to have it there. But clearly we both use the binary log as the main format.

A solution if we don't want to fully support two formats at the same time (which I think is a good idea) is to propose CSV as an export option only - make it clear in the API that there is a 'writeLog/readLog' pair for Jiminy's binary format, and an 'exportToCSV' tool for people who need it - that would be for interfacing with other tools only, standard jiminy project would use the binary file. What do you think ?

Reply:
I didn't mean to add support in C++, but at least in Python. We should ad an helper to read log in `log.py`. It would be only 2 lines or code using the native method `csv.DictReader` https://docs.python.org/3/library/csv.html#csv.DictReader.

Then, it would be great to have an optional parameter `is_binary` to your method (as for `write_log`) switching because the 2 reading methods.

Reply:
I agree it was present before your PR, but I think it does not require a lot of work but may greatly improve the support the csv format.

Reply:
Agreed, doing it in python is really fast so it makes sense. I was thinking about doing it in C++ at first because `Engine.read_log` is in C++.

How about I write a ```jiminy_py.read_log(logfile)``` function in python, that detects if the log is csv or binary (shouldn't be hard to do it automatically), and calls the right function accordingly ? I would rename the orginal, C++-binded method to ```Engine.read_log_binary``` to avoid confusion.

Reply:
It would be amazing :)  (or perhaps `jiminy_py.log.read_log` but whatever)

Reply:
Have a look at this https://stackoverflow.com/a/11301631/4820605

Reply:
I did things a bit differently based on other examples: I try to read the log as text and if a Unicode parsing exception is raised, I switch to binary. But I can implement your solution if you prefer.

Reply:
I have to admit I'm not a huge fun of `try ... catch ...` mechanism when a more intelligible criterion can be used. Moreover here the criterion is robust and easy to implement.

Reply:
From the comment in the code ('    FIXME: This approach incorrectly reports UTF-16 as binary.') it doesn't look super robust, but the try catch clearly wasn't either. I've changed to the proposed solution.

Reply:
Yes it is not, in a sense that it cannot be used in the general case, but we now what we expect to be a binary log file in our case, and what this method is checking will never change in the future. That's why I said it would robust :)